llvm/tools/llvm-foreach/llvm-foreach.cpp

```diff
@@ -79,6 +89,31 @@ static void error(std::error_code EC, const Twine &Prefix) {
     error(Prefix + ": " + EC.message());
 }
 
+// With WaitUntilTerminates=false this function just goes through the all
+// submitted jobs to check if one of them has finished.
+// TODO: give a proper naming as this function doesn't wait for jobs with
+// WaitUntilTerminates=false
```

Comment:
I suggest to rename the function into `checkIfJobsAreFinished` and rename `WaitUntilTerminates` to `BlockingWait`.

I.e. we call it without `BlockingWait` to see if _at least one_ job has finished, so we can submit another one and we call it with `BlockingWait` to actually wait for jobs completion